mobile/src/main/java/org/openhab/habdroid/ui/MainActivity.java

```diff
@@ -297,6 +297,10 @@ private void queryServerProperties() {
                 }
             }
         };
+        if (getConnection() == null) {
+            mController.indicateNoNetwork(getString(R.string.error_network_not_available));
+            return;
+        }
```

Comment:
This needs more thought. `queryServerProperties` currently assumes `mConnection` being non-null, and so does `retryServerPropertyQuery`. If mConnection is null and `ConnectionFailureFragment` is still shown (otherwise you can't click on it), there must be a case where we failed to update UI in `onAvailableConnectionChanged`, which is where the fix should be.